crates/libs/metadata/src/reader/mod.rs

```diff
@@ -1235,6 +1234,22 @@ impl<'a> Reader<'a> {
 
         SignatureKind::ReturnVoid
     }
+    fn signature_param_is_query_guid(&self, params: &[SignatureParam]) -> Option<usize> {
+        for pos in (0..params.len()).rev() {
```

Comment:
I think you're re-inventing [`Iterator::position`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.position).

Reply:
Except that I'm looking for the last match. I guess I could do `iter().rev().position(...)`.

Reply:
lol how did I miss that one...